### PR TITLE
Change mutation_rounds array members to proper objects.

### PR DESCRIPTION
**Pull request by SpencerMichaels**
This PR provides a minor fix suggested by @kevingranade in #22494. Previously, the extended list of mutation rounds in an NPC class would be specified with its members being arrays, as in

```
       "mutation_rounds" : [
           [ MUTCAT_LIZARD", { "constant" : 15 } ],
           [ "MUTCAT_FELINE", { "constant" : 15 } ]
       ]
```

 With this PR, they are changed to the following format.
```
       "mutation_rounds" : [
           { "category": "MUTCAT_LIZARD", "rounds": { "constant" : 15 } },
           { "category": "MUTCAT_FELINE", "rounds": { "constant" : 15 } }
       ]

```

### CODE REVIEW

**Comment by kevingranade** [comment]
I was thinking something like this, sorry I was unclear
```
"mutation_rounds": [
  { "MUTCAT_LIZARD": { "constant": 15 } },
  { "MUTCAT_ANY": { "constant": 15 } }
]
```
the code to parse this will look something like:
```
         while( jarr.has_more() ) {
            JsonObject jo2 = jarr.next_object();
            JsonObject distr_jo = jo2.get_object( "rounds" );
            mutation_rounds[ jo2.get_name() ] = load_distribution( distr_jo );
         }
```

**Comment by sxmichaels** [comment]
Oh, I see. Would it be better to format it like the following? It occurs to me that your format is somewhat ambiguous in that each object in the array could have more than one mutation tag in it (there is no `get_name()`, only `get_member_names()`). It looks like you could do it as one object, actually.

```
"mutation_rounds": {
    "MUTCAT_LIZARD" : { "constant": 15 },
    "MUTCAT_ANY" : { "constant": 15 },
}
```

**Comment by kevingranade** [comment]
That's a better syntax, but you were using the distinction between object and array to decide which kind of object it was, and I didn't see a way to do that cleanly without additional keywords or similar, so an array of objects seemed like the simplest option.

If you have a better solution, go for it.

**Comment by sxmichaels** [comment]
Oh, hmm, I hadn't thought of that. What do you think about removing the shortcut format entirely — just sticking to the long-form one — and then using that object notation in my last post? That's more consistent, anyway.

**Comment by kevingranade** [comment]
Now that you mention it that's better all around.

**Comment by sxmichaels** [comment]
Okay, I changed to that format then — no exception/shortcut case, with a pure object whose member names specify mutations. I also removed the need to add a `MUTCAT_` prefix, which is now automatically inserted, so the new format is like:

```
       "mutation_rounds" : {
           "SPIDER": { "constant" : 15 }
       }
```

File: src/mutation.h [hunk 1]
```diff
@@ -239,6 +239,7 @@ struct mutation_category_trait {
 
 void load_mutation_category( JsonObject &jsobj );
 void load_dream( JsonObject &jsobj );
+bool mutation_category_is_valid( std::string &cat );
```
Comment (BevapDin) [inline src/mutation.h]: Why does this require a *non-const* reference?

File: src/npc_class.cpp [hunk 2]
```diff
@@ -244,23 +244,22 @@ void npc_class::load( JsonObject &jo, const std::string & )
...
-        JsonArray jarr = jo.get_array( "mutation_rounds" );
-        while( jarr.has_more() ) {
-            JsonArray jarr2 = jarr.next_array();
-            JsonObject jo2 = jarr2.get_object( 1 );
-            mutation_rounds[jarr2.get_string( 0 )] = load_distribution( jo2 );
+        auto jo2 = jo.get_object( "mutation_rounds" );
+        for( auto &mutation : jo2.get_member_names() ) {
+            auto mutcat = "MUTCAT_" + mutation;
+            if( mutation_category_is_valid( mutcat ) ) {
+                debugmsg( "Unrecognized mutation category %s (i.e. %s)", mutation, mutcat );
```
Comment (BevapDin) [inline src/npc_class.cpp]: If it *is* valid, you show a debug message? Is there a `!` missing in the condition?

Comment (sxmichaels) [reply src/npc_class.cpp]: Oh gosh, that's embarrassing. I had just noticed and fixed that but forgot to commit before I pushed… fixed now!

**Pull request merged**